internal/impl/babysitter.go

```diff
@@ -211,6 +228,10 @@ func (b *babysitter) watchPods(ctx context.Context, component string) error {
 					delete(addrs, pod.Name)
 					changed = true
 				}
+			case watch.Error:
+				errObject := apierrors.FromObject(event.Object)
+				logger.Error("watcher error", "err", errObject)
+				// do nothing else? do we need? even with retryWatcher?
```

Comment:
no, we don't need to do anything, just remove

Reply:
isn't it safer to handle "unexpected" errors that the RetryWatcher might not be handling?
or maybe, just return an error here? 

because the channel will stop to send events. for me, not handling errors here, might still cause the service to "hang" forever.

Reply:
another option, could be to add the `watcher.Done()` to the select, like this:

```go
		select {
//
// other cases...
//
		case <-watcher.Done():
			return fmt.Errorf("watcher done")


		case event, ok := <-watcher.ResultChan():
```

And panic, or re-open the watcher from the outside.